library/src/main/java/org/xmtp/android/library/Conversations.kt

```diff
@@ -320,84 +296,156 @@ data class Conversations(
         return conversation
     }
 
+    suspend fun listGroups(
+        after: Date? = null,
+        before: Date? = null,
+        limit: Int? = null,
+        order: ConversationOrder = ConversationOrder.CREATED_AT,
+    ): List<Group> {
+        val ffiGroups = libXMTPConversations?.listGroups(
+            opts = FfiListConversationsOptions(
+                after?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                before?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                limit?.toLong()
+            )
+        ) ?: throw XMTPException("Client does not support V3 dms")
+
+        val groups = ffiGroups.map {
+            Group(client, it)
+        }
+
+        return when (order) {
+            ConversationOrder.LAST_MESSAGE ->
+                groups.sortedByDescending { group ->
+                    group.messages(limit = 1).firstOrNull()?.sent
+                }
```

Comment:
Allow ordering by last message to get rid of the need for batch messages? Also open to removing this from here and just letting integrators handle the ordering now that its much more performant?

Reply:
I think it might make sense to add a `last_message_ns` field to groups in libxmtp, which would be useful for multiple purposes